cobc/codegen.c

```diff
@@ -12601,9 +12621,8 @@ output_internal_function (struct cb_program *prog, cb_tree parameter_list)
 	}
 
 	seen = 0;
-	for (m = literal_cache; m; m = m->next) {
-		if (CB_TREE_CLASS (m->x) == CB_CLASS_NUMERIC
-		 && m->make_decimal) {
+	for (m = prog->decimal_constants; m; m = m->next) {
+		if (m->make_decimal) {
```

Comment:
all entries in that list have the decimal marker, so that additional check can be dropped, same below